src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

```diff
@@ -198,9 +198,14 @@ private static MemberDeclarationSyntax RemoveAllMembers(MemberDeclarationSyntax
             TypeDeclarationSyntax typeDeclaration;
             if (namedType.IsRecord)
             {
-                typeDeclaration = namedType.TypeKind is TypeKind.Class
-                    ? SyntaxFactory.RecordDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken())
-                    : SyntaxFactory.RecordStructDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
+                var isRecordClass = namedType.TypeKind is TypeKind.Class;
+                var declarationKind = isRecordClass ? SyntaxKind.RecordDeclaration : SyntaxKind.RecordStructDeclaration;
+                var classOrStructKeyword = SyntaxFactory.Token(isRecordClass ? default : SyntaxKind.ClassKeyword);
```

Comment:
I'm on my phone, so might have missed something, but shouldn't this be `SyntaxKind.StructKeyword`? Or does specifying the syntax kind mean it will automatically use a strict keyword? But if so, why specify the class keyword?

Reply:
> I'm on my phone, so might have missed something, but shouldn't this be `SyntaxKind.StructKeyword`? Or does specifying the syntax kind mean it will automatically use a strict keyword? But if so, why specify the class keyword?

I agree, it looks like it should have been StructKeyword. Also needs tests if possible.

Reply:
Thanks. We don't yet have any scenarios that need generation of record structs, so this will get coverage when we do.